neqo-transport/src/cid.rs

```diff
@@ -510,8 +510,18 @@ impl ConnectionIdManager {
     pub fn retire(&mut self, seqno: u64) {
         // TODO(mt) - consider keeping connection IDs around for a short while.
 
-        self.connection_ids.retire(seqno);
-        self.lost_new_connection_id.retain(|cid| cid.seqno != seqno);
+        let zero_len_cid = seqno == CONNECTION_ID_SEQNO_EMPTY
+            || self
+                .connection_ids
+                .cids
+                .iter()
+                .any(|c| c.seqno == seqno && c.cid.len() == 0);
```

Comment:
This pull request adds to checks for zero-len-cid, here and in `ConnectionIdStore::retire_prior_to`. How about making this a method on `ConnectionIdEntry` instead, which is then called here and in `ConnectionIdStore::retire_prior_to`?